backend/app/project_handler.go

```diff
@@ -0,0 +1,152 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"gorm.io/gorm"
+)
+
+type createProjectInput struct {
+	Name string `json:"name" binding:"required"`
+}
+
+type updateProjectInput struct {
+	OwnerId uint   `json:"owner_id" binding:"required"`
+	Name    string `json:"name" binding:"required"`
+}
+
+func (a *App) createProject(ctx *gin.Context) (interface{}, Response) {
+	var input createProjectInput
+
+	if err := ctx.BindJSON(&input); err != nil {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("failed to read project data"))
+	}
+
+	// TODO: get user id from authorization middleware and assign it to OwnerId
+	newProject := models.Project{Name: input.Name, OwnerId: 10007} // 10007 is just a random number
+
+	// check if project name exists before
+	_, err := a.client.GetProjectByName(input.Name) // expected to return 'gorm.ErrRecordNotFound' if not exist
+	if err == nil {                                 // project is found by name
+		return nil, BadRequest(errors.New("project name must be unique"))
+	}
+	if err != gorm.ErrRecordNotFound { // there is some error other than not found
+		log.Error().Err(err).Send()
+		return nil, InternalServerError(errors.New("failed to create project"))
```

Comment:
I think we should return `internal server error` as a msg for all InternalServerErrors